src/memory.cpp

```diff
@@ -33,12 +33,18 @@ inline namespace CUPLA_ACCELERATOR_NAMESPACE
     CUPLA_HEADER_ONLY_FUNC_SPEC
     cuplaError_t cuplaMalloc(void** ptrptr, size_t size)
     {
-        const ::alpaka::Vec<cupla::AlpakaDim<1u>, cupla::MemSizeType> extent(size);
-
-        auto& buf = cupla::manager::Memory<cupla::AccDev, cupla::AlpakaDim<1u>>::get().alloc(extent);
+        try
+        {
+            const ::alpaka::Vec<cupla::AlpakaDim<1u>, cupla::MemSizeType> extent(size);
+            auto& buf = cupla::manager::Memory<cupla::AccDev, cupla::AlpakaDim<1u>>::get().alloc(extent);
 
-        // @toto catch errors
-        *ptrptr = ::alpaka::getPtrNative(buf);
+            *ptrptr = ::alpaka::getPtrNative(buf);
+        }
+        catch(...)
+        {
+            *ptrptr = nullptr;
+            return cuplaErrorMemoryAllocation;
+        }
```

Comment:
I think we should add a short note about the (new) behavior when allocation failed - that the pointer is null and return code is `cuplaErrorMemoryAllocation`.